stacks-signer/src/signerdb.rs

```diff
@@ -523,10 +611,36 @@ impl SignerDb {
         Ok(())
     }
 
+    /// Migrate from schema 4 to schema 5
+    fn schema_5_migration(tx: &Transaction) -> Result<(), DBError> {
+        if Self::get_schema_version(tx)? >= 5 {
+            // no migration necessary
+            return Ok(());
+        }
+
+        for statement in SCHEMA_5.iter() {
+            tx.execute_batch(statement)?;
+        }
+
+        Ok(())
+    }
+
     /// Either instantiate a new database, or migrate an existing one
     /// If the detected version of the existing database is 0 (i.e., a pre-migration
     /// logic DB, the DB will be dropped).
     fn create_or_migrate(&mut self) -> Result<(), DBError> {
+        // Register helper function for determining if a block is a tenure change transaction
+        self.db.create_scalar_function(
```

Comment:
Instead of registering a user-defined sqlite function, why not just add a column with this data?  It'll be faster to retrieve that way.

Reply:
I did add a column. This enables me to migrate the data. the column name is tenure_change but in order to extract the data during the migration, I need this function since its not part of block info and even if I add it to block info after the fact, it won't exist for old data.

Reply:
Ah, I see.

You should de-register the function after migration if it's not going to be used afterwards.

Reply:
Alright! I will do!